_posts/2024-06-18-livechatpro.md

```diff
@@ -0,0 +1,75 @@
+---
+layout: post
+title: "[CVE-2024-36679] Improper Control of Generation of Code in Module Live Chat Pro (All in One Messaging) module for PrestaShop"
+categories: modules
+author:
+- Touchweb.fr
+- 202 ecommerce.com
+meta: "CVE,PrestaShop,livechatpro"
+severity: "critical (10.0)"
+---
+
+In the module "Module Live Chat Pro (All in One Messaging)" (livechatpro), a guest can perform PHP Code injection in affected versions.
+
+## Summary
+
+* **CVE ID**: [CVE-2024-36679](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2024-36679)
+* **Published at**: 2024-06-18
+* **Advisory source**: Friends-Of-Presta.org
+* **Platform**: PrestaShop
+* **Product**: livechatpro
+* **Impacted release**: <= 8.4.0 (see WARNING below)
+* **Product author**: ProQuality
+* **Weakness**: [CWE-94](https://cwe.mitre.org/data/definitions/94.html)
+* **Severity**: critical (10.0)
+
+## Description
+
+Due to a predictable token, the method `Lcp::saveTranslations()` suffer of a white writer that can inject PHP code into a PHP file which will lead to critical RCE.
+
+**WARNING** : Author refuse to patch the vulnerability so you should consider to uninstall it. There is strong design issue which cannot be fix by a hotfix. Version tagged as impacted is the only version we had time to produce a POC it, author has updated things in newer versions but its token is still predictible. So you should consider that all versions are impacted.
```

Comment:
Version tagged as impacted is the only version we had time to produce a POC **for**,